### PR TITLE
implemented HUD, asteroid size, minor gameplay tweaks

### PR DESCRIPTION
**Pull request by LeastCreative**

### CODE REVIEW

File: asteroid.js [hunk 1]
```diff
@@ -37,13 +48,16 @@ function Asteroid(pos, r) {
...
+      return [new Asteroid(this.pos, this.r, this.size-1), new Asteroid(this.pos, this.r, this.size-1)];
+    else
+      return [];
   }
 
   this.vertices = function() {
     var vertices = []
     for(var i = 0; i < this.total; i++) {
-      var angle = this.heading + map(i, 0, this.total, 0, TWO_PI);
+      var angle = map(i, 0, this.total, 0, TWO_PI);
```
Comment (meiamsome) [inline asteroid.js]: This incorrectly reverts a previous commit.

File: asteroid.js [hunk 2]
```diff
@@ -3,22 +3,33 @@
...
-function Asteroid(pos, r) {
-  if (pos === undefined) {
-    pos = createVector(random(width), random(height))
+function Asteroid(pos, r, size) {
+  if (!pos) {
+    pos = createVector(random(width), random(height));
   }
 
-  r = r !== undefined ? r * 0.5 : random(15, 30);
+  r = r ? r * 0.5 : random(40, 60);
```
Comment (MunWolf) [inline asteroid.js]: this will not work if the radius is 0 but I guess that is fine since having a 0 as a radius is bad anyway.

File: hud.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                break;
+            default:
+                console.log("line map is invalid");
+                break;
+        }
+
+    }
+
+
+
```
Comment (MunWolf) [inline hud.js]: maybe clean up all the extra spaces in this commit?
Also the indents are bigger then the other files.
Sorry i'm nitpicking here.

File: index.html [hunk 4]
```diff
@@ -1,16 +1,23 @@
...
+    <script language="javascript" type="text/javascript" src="hud.js"></script>
+    <script language="javascript" type="text/javascript" src="asteroid.js"></script>
+    <script language="javascript" type="text/javascript" src="laser.js"></script>
+    <style>
+        body {
+            padding: 0;
+            margin: 0;
+        }
+    </style>
 </head>
```
Comment (MunWolf) [inline index.html:14]: Maybe not change the indentation on everything so it doesn't go all green and red on us.

Comment (LeastCreative) [reply index.html:14]: AH! sorry. I set my formatting to match everyone else on JS but I guess the HTML one is still default

Comment (MunWolf) [reply index.html:14]: No problem mate, that's what reviewing is for :)

**Pull request closed without merging**